internal/service/apigateway/domain_name.go

```diff
@@ -290,6 +294,7 @@ func resourceDomainNameRead(ctx context.Context, d *schema.ResourceData, meta in
 	d.Set("cloudfront_zone_id", meta.(*conns.AWSClient).CloudFrontDistributionHostedZoneID(ctx))
 	d.Set(names.AttrDomainName, output.DomainName)
 	d.Set("domain_name_id", output.DomainNameId)
+	d.Set("domain_name_arn", output.DomainNameArn)
```

Comment:
using `arn` doesnt work in upstream APIs. I dont want to break any existing assumptions but it seems to be incorrect so I added the new param. I dont mind removing this and setting the `arn` to expected value. up to maintainers

Reply:
Let's try a stick with `arn`.
According to the documentation (e.g. https://docs.aws.amazon.com/apigateway/latest/api/API_GetDomainName.html#apigw-GetDomainName-response-domainNameArn) `output.DomainNameArn` is only set for private custom domain names.
We could do something like

```go
	if output.DomainNameArn != nil {
		d.Set(names.AttrARN, output.DomainNameArn)
	} else {
		d.Set(names.AttrARN, domainNameARN(ctx, meta.(*conns.AWSClient), d.Id()))
	}
```

Reply:
And remove the new `domain_name_arn` attribute.

Reply:
I am working on this...